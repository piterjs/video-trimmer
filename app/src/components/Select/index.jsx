import React from 'react';

import './Select.css';
import './colors/default.css';
import './colors/danger.css';
import './colors/info.css';
import './colors/success.css';
import './colors/warning.css';

import '../Control/Control.css';
import '../Control/sizes.css';

const Select = ({size = 'normal', color = 'default', className = '', ...rest}) => {
    const classes = [
        'Control',
        `Control--size-${size}`,
        'Select',
        `Select--color-${color}`,
        className
    ].join(' ')

    return <select
        className={classes}
        {...rest}
    >{rest.children}</select>
};

export default Select;
