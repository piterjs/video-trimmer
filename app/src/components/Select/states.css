.Select[multiple] {
    height: 2.25em;
}
