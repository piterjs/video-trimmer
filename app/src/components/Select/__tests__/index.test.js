import React from 'react';
import renderer from 'react-test-renderer';
import Select from '../';

describe('Select', () => {
    it('should be defined', () => {
        expect(Select).toBeDefined();
    });
    it('should render correctly', () => {
        const tree = renderer
        .create(<Select><option value="test">Test</option></Select>)
        .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
