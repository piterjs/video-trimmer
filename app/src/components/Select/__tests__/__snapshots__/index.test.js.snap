// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render correctly 1`] = `
<select
  className="Control Control--size-normal Select Select--color-default "
>
  <option
    value="test"
  >
    Test
  </option>
</select>
`;
